src/sdk/service/network-validation-service/__tests__/network-validation-service.test.unit.ts

```diff
@@ -0,0 +1,357 @@
+    [new IPv4Address('10.0.0.0')],
+    {
+      start: '10.0.0.10',
+      end: '10.0.0.255'
+    }
+  )).toBeFalsy();
+  expect(NetworkValidationService.isAnyIPv4AddressInIPv4AddressRange(
+    [
+      new IPv4Address('10.0.0.0'),
+      new IPv4Address('10.0.0.1'),
+      new IPv4Address('10.0.0.2')
+    ],
+    {
+      start: '10.0.0.10',
+      end: '10.0.0.255'
+    }
+  )).toBeFalsy();
+  expect(NetworkValidationService.isAnyIPv4AddressInIPv4AddressRange(
+    [
+      new IPv4Address('10.0.0.11'),
+      new IPv4Address('10.0.0.12'),
+      new IPv4Address('10.0.0.13')
+    ],
+    {
+      start: '10.0.0.10',
+      end: '10.0.0.255'
+    }
+  )).toBeTruthy();
+  expect(NetworkValidationService.isAnyIPv4AddressInIPv4AddressRange(
+    [
+      new IPv4Address('10.0.0.11'),
+      new IPv4Address('10.0.0.12'),
+      new IPv4Address('10.0.0.13')
+    ],
+    {
+      start: '10.0.0.10',
+      end: '10.0.0.255'
+    }
+  )).toBeTruthy();
+});
+
+test('Check if any element in array of IPv4 Addresses is in any of the ranges.', () => {
+  expect(NetworkValidationService.isAnyIPv4AddressInAnyIPv4AddressRange(
+    [new IPv4Address('10.0.0.11'), new IPv4Address('10.0.0.1'), new IPv4Address('10.0.0.3')],
+    [{
+      start: '10.0.0.10',
+      end: '10.0.0.255'
+    }]
+  )).toBeTruthy();
+  expect(NetworkValidationService.isAnyIPv4AddressInAnyIPv4AddressRange(
+    [new IPv4Address('10.0.0.11'), new IPv4Address('10.0.0.12'), new IPv4Address('10.0.0.13')],
+    [{
+      start: '10.0.0.1',
+      end: '10.0.0.10'
+    }]
+  )).toBeFalsy();
+  expect(NetworkValidationService.isAnyIPv4AddressInAnyIPv4AddressRange(
+    [new IPv4Address('10.0.0.11'), new IPv4Address('10.0.0.12'), new IPv4Address('10.0.0.13')],
+    [{
+      start: '10.0.0.10',
+      end: '10.0.0.255'
+    },
+    {
+      start: '10.0.0.1',
+      end: '10.0.0.10'
+    }]
+  )).toBeTruthy();
+});
+
+test('Check if IPv4 Address fits within any ranges in array.', () => {
+  expect(NetworkValidationService.isIPv4AddressInAnyIPv4AddressRange(
+    new IPv4Address('10.0.0.11'),
+    [{
+      start: '10.0.0.1',
+      end: '10.0.0.10'
+    },
+    {
+      start: '10.0.0.11',
+      end: '10.0.0.15'
+    }]
+  )).toBeTruthy();
+  expect(NetworkValidationService.isIPv4AddressInAnyIPv4AddressRange(
+    new IPv4Address('10.0.0.10'),
+    [{
+      start: '10.0.0.1',
+      end: '10.0.0.9'
+    },
+    {
+      start: '10.0.0.11',
+      end: '10.0.0.15'
+    }]
+  )).toBeFalsy();
+});
+
+test('Check if gets IPv4 Address as integer and vice-versa.', () => {
+  expect(NetworkValidationService.getIntegerAsIPv4Address(-1062731775)).toEqual(
+    '192.168.0.1'
+  );
+  expect(NetworkValidationService.getIPv4AsInteger(new IPv4Address('192.168.0.1'))).toEqual(
+   -1062731775
+  );
+});
+
+
+test('Check if getNetmaskPrefixSize function gets Netmask prefix size.', () => {
+  expect(NetworkValidationService.getNetmaskPrefixSize(new Netmask('255.255.255.0'))).toEqual(
+    24
```

Comment:
check a few more, especially the extremes (0, 32). You generally always want to at least test the extreme cases